File: 121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        min_cost = prices[0]

        for price in prices:
            min_cost = min(min_cost, price)
            max_profit = max(max_profit, price - min_cost)

        return max_profit

File: 121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int low = INT_MAX, max_profit = 0;
        for(int i = 0; i < prices.size(); i++) {
            if(prices[i] < low) {
                low = prices[i];
            }
            max_profit = fmax(max_profit, prices[i] - low);
        }
        return max_profit;
    }
};